### PR TITLE
Make CVMFS fix configurable

### PR DESCRIPTION
**Pull request by NilsKrattinger**
Add a option to not deploy the CVMFS-fix. 
~Add ability to restart nodeplugin even if deployed on other namespace~ 
Add condition to not create the cvmfs-fix configmap if CVMFS isn't used / deployed as CSI for refdata. 

Close [#494](https://github.com/galaxyproject/galaxy-helm/issues/494).
~Impact #441.~

### CODE REVIEW

File: galaxy/templates/rbac-fix-cvmfs.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+rules:
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "list","watch","delete"]
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ include "galaxy.fullname" . }}-cvmfs-fix
+  namespace: {{ .Values.cvmfs.externalCvmfsNamespace }}
```
Comment (nuwang) [inline galaxy/templates/rbac-fix-cvmfs.yaml:31]: In general, I think it's preferable to constrain the helm chart to a single namespace. We don't want to expand the security footprint where it reaches across to other namespaces. Therefore, I think that if someone wants to create this service account, they do so manually, and we just document it.

**Review by nuwang — COMMENTED** [review]
Thanks for this fix @NilsKrattinger. My only suggestion is that we add the `deployPostInstallFix` flag only, and remove the `externalCvmfsNamespace` option. This fix was never meant to be permanent, just a temporary workaround till the upstream csi chart is fixed. I think it's preferable to simply document the steps or manually restart the csi pods instead.

**Comment by NilsKrattinger** [comment]
@nuwang, I have removed the ability to restart the external Nodepluing CVMFS. 

Changes:
- Add a option to not deploy the CVMFS-fix. ``
- Add condition to not create the cvmfs-fix configmap if CVMFS isn't used / deployed as CSI for refdata.

Close #494.

**Comment by nuwang** [comment]
@NilsKrattinger Thanks for the changes. @ksuderman is also investigating whether this may have been fixed upstream, and hopefully he has some good news to share.

**Pull request merged**